ansible/roles/test/files/ptftests/arista.py

```diff
@@ -313,6 +331,49 @@ def parse_bgp_route(self, output, expects):
 
         return set(expects) == prefixes
 
+    def get_bgp_neighbors(self):
+        neigh_bgp = {}
+        dut_bgp = {}
+        for cmd, ver in [('show ip bgp neighbors', 'v4'), ('show ipv6 bgp neighbors', 'v6')]:
+            output = self.do_cmd(cmd)
+            if ver == 'v6':
+                neigh_bgp[ver], dut_bgp[ver], neigh_bgp['asn'] = self.parse_bgp_neighbor(output, get_neigh=True, get_asn=True)
+            else:
+                neigh_bgp[ver], dut_bgp[ver] = self.parse_bgp_neighbor(output, get_neigh=True)
+
+        return neigh_bgp, dut_bgp
+
+    def change_bgp_neigh_state(self, asn, state="no shut"):
```

Comment:
>state [](start = 42, length = 5)

better use a bool, such as is_up?

Reply:
Had it this way since I use the state in my command directly to shut/no shut the bgp neighbors. if I use a bool, then I would need a mapping for false->shut, true -> no shut

Reply:
My point here is the possibility to inject invalid string from function parameter. If you prefer string, you need to validate the string before exec it.

---
In reply to: [287446148](https://github.com/Azure/sonic-mgmt/pull/926#discussion_r287446148) [](ancestors = 287446148)

Reply:
Makes sense. I will use a bool and then map it to my command in the function.

Reply:
`state="no shut"` is still there.

---
In reply to: [287534126](https://github.com/Azure/sonic-mgmt/pull/926#discussion_r287534126) [](ancestors = 287534126)

Reply:
Made the change. Haven't pushed it yet. Testing out some changes

Reply:
Updated the change